internal/attestations/attestations.go

```diff
@@ -50,9 +51,16 @@ type Attestations struct {
 	// `<ref-path>/<from-id>-<to-id>`, where `ref-path` is the absolute ref path
 	// such as `refs/heads/main` and `from-id` and `to-id` determine how the ref
 	// in question moved. For example, the key
-	// `refs/heads/main/<commit-A>-<commit-B>` indicates the authorization is
-	// for the action of moving `refs/heads/main` from `commit-A` to `commit-B`.
+	// `refs/heads/main/<commit-A>-<tree-B>` indicates the authorization is
+	// for the action of moving `refs/heads/main` from `commit-A` to a commit
+	// with `tree-B`.
 	referenceAuthorizations map[string]plumbing.Hash
+
+	// githubPullRequestAttestations maps information about each GitHub pull
+	// request. The key is a path of the form `<ref-path>/<from-id>-<to-id>`,
```

Comment:
I don't know if this makes sense for pull requests 🤔 

`<ref-path>`: Is this the base ref or the PR ref? The changes are being made on the PR ref, but what if this ref doesn't exist in the base repo?

`<from-id>`: How do you see the from-id being used? I'm a bit worried that this might mirror the "Require branches to be up to date before merging" behavior for branch protections, which many repos end up turning off. Would it make more sense to key attestation off the PR commit and perform any base commit checks from metadata stored in the attestation?

Also idk how often this happens in practice, but will this be problematic if there is a collision? I think GitHub does try to get you to reuse PRs if there's already one open, but you can have multiple closed pull requests that point to the same refs/commits (e.g. https://github.com/wlynch/test/pull/29, https://github.com/wlynch/test/pull/30). An attestation for the closed PR could bleed into the open one.

Reply:
I've updated to store as follows.

a) It's still in refs/gittuf/attestations
b) There's still a github-pull-request dir in that tree
c) The path to the attestation is <base_branch>/<merge_sha>

If I check out that ref, I see `github-pull-requests/refs/heads/main/b69a55e72103a1560b5dbe2e4c961b488290a4ac` for example. The subject of that attestation is also now using the `gitCommit` digest instead of the `gitTree` digest.

Reply:
> With PR commit, do you mean basically store this keyed by the PR merge / squash commit?

I was thinking the PR head commit so that PR attestations any time, similar to status checks. Using merge commits are tricky because they'll change constantly.

If the short term goal is to store attestations about PRs at merge time, then could this be stored in the existing ref attestation type with a different predicate?

Maybe we need 2 predicates? One to store the PR data on the PR commit and one to link the merge to the PR?

Reply:
I don't know that we need different predicate types as the contents and really the intent to is basically the same. I think I can tweak this so it can be configured to be recorded on merge as well as on push in a pull request.

Here's what I suggest. If `merged_at` is not null, meaning the PR is merged, we store against the `merge_sha`. Otherwise, we store against the head of the feature branch. That should open this up to being recorded on push to `main` as well as on pull_request.

Reply:
I've updated this now. If the PR is merged, the attestation is stored against the merge_sha and the base branch. If the PR is not yet merged, it's stored against the HEAD of the feature branch. In both cases, I also started recording the username + immutable ID, we need this for feature branches to know where they came from.